software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -20,64 +20,111 @@
+        setAttribute(SLAVES, slaves);
+
+        addEnricher(Enrichers.builder()
+                .propagating(RedisStore.HOSTNAME, RedisStore.ADDRESS, RedisStore.SUBNET_HOSTNAME, RedisStore.SUBNET_ADDRESS, RedisStore.REDIS_PORT)
+                .from(master)
+                .build());
+    }
+
+    @Override
+    protected void initEnrichers() {
+        super.initEnrichers();
+        ServiceStateLogic.newEnricherFromChildrenUp().
+            checkChildrenOnly().
+            requireUpChildren(QuorumChecks.all()).
+            configure(ComputeServiceIndicatorsFromChildrenAndMembers.IGNORE_ENTITIES_WITH_THESE_SERVICE_STATES, ImmutableSet.<Lifecycle>of()).
+            addTo(this);
     }
     
     @Override
     public void start(Collection<? extends Location> locations) {
-        master = addChild(EntitySpec.create(RedisStore.class));
-        Entities.manage(master);
-        master.start(locations);
+        ServiceStateLogic.setExpectedState(this, Lifecycle.STARTING);
+        ServiceProblemsLogic.clearProblemsIndicator(this, START);
+        try {
+            doStart(locations);
+            ServiceStateLogic.setExpectedState(this, Lifecycle.RUNNING);
+        } catch (Exception e) {
+            ServiceProblemsLogic.updateProblemsIndicator(this, START, "Start failed with error: "+e);
+            ServiceStateLogic.setExpectedState(this, Lifecycle.ON_FIRE);
+            throw Exceptions.propagate(e);
+        }
+    }
 
-        slaves = addChild(EntitySpec.create(DynamicCluster.class)
-                .configure(DynamicCluster.MEMBER_SPEC, EntitySpec.create(RedisSlave.class).configure(RedisSlave.MASTER, master)));
-        slaves.start(locations);
+    private void doStart(Collection<? extends Location> locations) {
+        RedisStore master = getMaster();
+        master.invoke(RedisStore.START, ImmutableMap.<String, Object>of("locations", ImmutableList.copyOf(locations))).getUnchecked();
 
-        setAttribute(Startable.SERVICE_UP, calculateServiceUp());
+        DynamicCluster slaves = getSlaves();
+        slaves.invoke(DynamicCluster.START, ImmutableMap.<String, Object>of("locations", ImmutableList.copyOf(locations))).getUnchecked();
     }
 
     @Override
     public void stop() {
-        if (slaves != null) slaves.stop();
-        if (master != null) master.stop();
+        ServiceStateLogic.setExpectedState(this, Lifecycle.STOPPING);
+        try {
+            doStop();
+            ServiceStateLogic.setExpectedState(this, Lifecycle.STOPPED);
+        } catch (Exception e) {
+            ServiceProblemsLogic.updateProblemsIndicator(this, STOP, "Stop failed with error: "+e);
+            ServiceStateLogic.setExpectedState(this, Lifecycle.ON_FIRE);
+            throw Exceptions.propagate(e);
+        }
+    }
 
-        setAttribute(Startable.SERVICE_UP, false);
+    private void doStop() {
+        getSlaves().invoke(DynamicCluster.STOP, ImmutableMap.<String, Object>of()).getUnchecked();
```

Comment:
This is fine for the PR, but I suspect if there is an error stopping the slaves then the master won't be stopped either. Not sure off-hand what best approach is for that.